c++/src/Reader.cc

```diff
@@ -1433,6 +1538,7 @@ namespace orc {
   }
 
   void ReaderImpl::startNextStripe() {
+    reader.reset(); // ColumnReaders use lots of memory; free old memory first
```

Comment:
Why? On line 1550, we assign reader a new value. It destroys the previous value anyway. What calling reset() explicitly does for us is it frees up all the dynamic memory used in the existing instance of ColumnReader BEFORE creating a new ColumnReader and allocating all the buffers. Without calling reset(), the deallocation happens AFTER a new reader is created, so for a brief moment we use 2x memory than we really need. It messes up our memory estimates, too.